http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeService.scala

```diff
@@ -0,0 +1,277 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package http
+package service
+
+import java.time.Instant
+
+import cats.effect.{Async, ContextShift, IO, Timer}
+import cats.mtl.ApplicativeAsk
+import io.chrisdavenport.log4cats.Logger
+import org.broadinstitute.dsde.workbench.leonardo.db.{DbReference, SaveCluster, clusterQuery}
+import org.broadinstitute.dsde.workbench.leonardo.http.service.RuntimeService.CreateRuntimeRequest
+import org.broadinstitute.dsde.workbench.leonardo.model.google.{ClusterName, ClusterStatus}
+import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, ClusterRequest, DefaultLabels, LeoAuthProvider, MachineConfigOps, ProjectActions, RuntimeConfigRequest, ServiceAccountProvider}
+import org.broadinstitute.dsde.workbench.model.google.{GcsPath, GoogleProject}
+import org.broadinstitute.dsde.workbench.model.{TraceId, UserInfo, WorkbenchEmail}
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.leonardo.ClusterImageType.{Jupyter, Welder}
+import org.broadinstitute.dsde.workbench.leonardo.config.{AutoFreezeConfig, ClusterDefaultsConfig, DataprocConfig, ImageConfig}
+import org.broadinstitute.dsde.workbench.leonardo.dao.DockerDAO
+import LeonardoService._
+import org.broadinstitute.dsde.workbench.leonardo.model.Cluster.getClusterUrl
+import RuntimeService._
+import org.broadinstitute.dsde.workbench.google2.GoogleStorageService
+import org.broadinstitute.dsde.workbench.leonardo.http.api.RuntimeServiceAPIContext
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage
+import org.broadinstitute.dsde.workbench.model.google
+
+import scala.concurrent.ExecutionContext
+
+class RuntimeService[F[_]: ContextShift](config: RuntimeServiceConfig,
```

Comment:
Question: do we need a separate `RuntimeService` class? I was thinking `LeonardoService` would handle create/get/stop/start/delete runtime requests, and call the appropriate `Helper` methods depending on whether it's Dataproc or GCE.

Reply:
I started in `LeonardoService`, but then realize there will be lots of rewrite..and I didn't want to mess with dataproc route too much. I'm hoping to once things are more finalized in RuntimeService, we can slowly reorg the code in leonardoservice and runtimeservice a bit